modules/libcom/src/osi/os/posix/osdThread.c

```diff
@@ -326,13 +330,22 @@ int          status;
     a_p->usePolicy = arg.ok;
 }
 #endif
-
+
+static int childAfterFork;
```

Comment:
There could be an enum to declare the magic numbers that this variable will hold (currently 0, 1 and 2), making them self-documenting (e.g. `forkNone`, `forkDone`, `forkWarned`).

Reply:
How about a comment here listing states and meanings?

Switching to an enum requires casting back to `int*` for the epicsAtomic calls.  eg. 

```c
epicsAtomicSetIntT(&childAfterFork, 1);
// becomes
epicsAtomicSetIntT((int*)&childAfterFork, forkChild);
```

Reply:
My intention was to just use the enum to declare the magic number values (you could use `#define`s instead but I think an enum is slightly a better solution), but leave the `childAfterFork` variable as an `int` (with a comment that it should contain one of those enum values) — that seems to be the recommended way to do this kind of thing in C without having to do the casting of pointers to the variable. IIRC you can't guarantee what width of integer the compiler will actually use if you make it an `enum`.

Reply:
My reason for suggesting the enum was to document the magic number states lower down in the code. `epicsAtomicCmpAndSwapIntT(&childAfterFork, forkDone, forkWarned) == forkDone` is quicker to read and understand than `epicsAtomicCmpAndSwapIntT(&childAfterFork, 1, 2) == 1`.